File: scala/StalinSortSimple.scala
import scala.util.Random

object StalinSort {
  def stalinSort(list: List[Int]): List[Int] = {
    var max = Int.MinValue
    list.filter(v => if (v >= max) {
      max = v; true
    } else false)
  }
}

object Main extends App {
  (1 to 5).foreach { _ =>
    val list = List.fill(10)(Random.nextInt % 100)
    val sortedList = StalinSort.stalinSort(list)

    println("Original Array: " + list)
    println("Sorted: " + sortedList)
  }
}

File: python/stalin_sort_simple.py
def StalinSort(list):
    i = 0
    length = len(list)
    if length > 0:
        aux = list[0]
    while i < length and length > 0:
        if aux <= list[i] :
            aux = list[i]
            i = i+1
        else:
            del list[i]
            length = length -1

list = [1,2,3,5,4,6,4,8,9]
StalinSort(list)
print(list)